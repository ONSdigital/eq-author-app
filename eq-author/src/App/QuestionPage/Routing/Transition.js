import { CSSTransition } from "react-transition-group";
import PropTypes from "prop-types";
import styled from "styled-components";

const timeout = props => props.timeout;
const halfTimeout = props => props.timeout / 2;

const handleExit = node => {
  const { height } = node.getBoundingClientRect();
  node.style.height = `${height}px`;
};

const RoutingComponentTransition = styled(CSSTransition).attrs({
  classNames: "component",
  onExit: () => handleExit
})`
  position: relative;

  &.component-enter {
    opacity: 0;
    transform: scale(0.9);
    z-index: 200;
  }

  &.component-enter-active {
    opacity: 1;
    transform: scale(1);
    transition: opacity ${timeout}ms ease-out,
      transform ${timeout}ms cubic-bezier(0.175, 0.885, 0.32, 1.275);
  }

  &.component-exit {
    opacity: 1;
    transform: scale(1);
  }

  &.component-exit-active {
    opacity: 0;
    height: 0 !important;
    transform: scale(0.9);
    transition: opacity ${halfTimeout}ms ease-out,
      height ${halfTimeout}ms ease-in ${halfTimeout}ms,
      transform ${halfTimeout}ms ease-in;
  }
`;

RoutingComponentTransition.propTypes = {
  timeout: PropTypes.number,
  children: PropTypes.element
};

RoutingComponentTransition.defaultProps = {
  timeout: 200
};

export default RoutingComponentTransition;
