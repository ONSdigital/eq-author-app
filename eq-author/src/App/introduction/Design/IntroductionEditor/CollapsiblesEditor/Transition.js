import { CSSTransition } from "react-transition-group";
import PropTypes from "prop-types";
import styled from "styled-components";

const timeout = (props) => props.timeout;
const halfTimeout = (props) => props.timeout / 2;

const handleExit = (node) => {
  const { height } = node.getBoundingClientRect();
  node.style.height = `${height}px`;
};

const classNames = "transition";

const Transition = styled(CSSTransition).attrs(() => ({
  classNames,
  onExit: () => handleExit,
}))`
  position: relative;

  &.${classNames}-enter {
    opacity: 0;
    transform: scale(0.9);
    z-index: 200;
  }

  &.${classNames}-enter-active {
    opacity: 1;
    transform: scale(1);
    transition: opacity ${timeout}ms ease-out,
      transform ${timeout}ms cubic-bezier(0.175, 0.885, 0.32, 1.275);
  }

  &.${classNames}-exit {
    opacity: 1;
    transform: scale(1);
  }

  &.${classNames}-exit-active {
    opacity: 0;
    height: 0 !important;
    transform: scale(0.9);
    transition: opacity ${halfTimeout}ms ease-out,
      height ${halfTimeout}ms ease-in ${halfTimeout}ms,
      transform ${halfTimeout}ms ease-in;
  }
`;

Transition.propTypes = {
  timeout: PropTypes.number,
  children: PropTypes.element,
};

Transition.defaultProps = {
  timeout: 200,
};

export default Transition;
