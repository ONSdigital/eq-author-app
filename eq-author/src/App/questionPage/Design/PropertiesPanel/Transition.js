import styled from "styled-components";
import { CSSTransition } from "react-transition-group";
import PropTypes from "prop-types";

const timeout = props => props.timeout;
const halfTimeout = props => props.timeout / 2;

const onExit = node => {
  const { height } = node.getBoundingClientRect();
  node.style.height = `${height}px`;
};

const Transition = styled(CSSTransition).attrs({
  classNames: "transition",
  onExit: () => onExit
})`
  &.transition-enter {
    opacity: 0;
  }

  &.transition-enter-active {
    opacity: 1;

    transition: opacity ${timeout}ms ease-out;
  }

  &.transition-exit {
    opacity: 1;
  }

  &.transition-exit-active {
    opacity: 0;

    height: 0 !important;
    transition: opacity ${halfTimeout}ms ease-out,
      height ${halfTimeout}ms ease-in ${halfTimeout}ms;
  }
`;

Transition.defaultProps = {
  in: false,
  timeout: 300
};

Transition.propTypes = {
  timeout: PropTypes.number,
  children: PropTypes.element
};

export default Transition;
