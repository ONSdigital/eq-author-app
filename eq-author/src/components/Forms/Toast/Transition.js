import styled from "styled-components";
import { CSSTransition } from "react-transition-group";
import PropTypes from "prop-types";

const ToastTransition = styled(CSSTransition).attrs({
  classNames: "toast"
})`
  transition-property: transform, opacity;
  transition-duration: ${props => props.timeout * 0.66}ms;

  &.toast-enter {
    opacity: 0;
    transform: translateY(50%);
    transition-timing-function: cubic-bezier(0.73, 0.02, 0.32, 1.49);
    transition-delay: ${props => props.timeout * 0.33}ms;
  }

  &.toast-enter.toast-enter-active {
    opacity: 1;
    transform: translateY(0);
  }

  &.toast-exit {
    opacity: 1;
    transform: translateY(0);
    transition-timing-function: cubic-bezier(0.75, -0.44, 0.24, 1);
  }

  &.toast-exit.toast-exit-active {
    opacity: 0;
    transform: translateY(-50%);
  }
`;

ToastTransition.defaultProps = {
  in: false,
  timeout: 500
};

ToastTransition.propTypes = {
  timeout: PropTypes.number,
  children: PropTypes.element
};

export default ToastTransition;
