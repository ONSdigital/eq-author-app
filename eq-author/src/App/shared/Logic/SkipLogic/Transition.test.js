import React from "react";
import { shallow } from "enzyme";

import Transition from "./Transition";

describe("components/Routing/Transition", () => {
  it("should render", () => {
    expect(
      shallow(
        <Transition>
          <div>Content</div>
        </Transition>
      )
    ).toMatchSnapshot();
  });
});
