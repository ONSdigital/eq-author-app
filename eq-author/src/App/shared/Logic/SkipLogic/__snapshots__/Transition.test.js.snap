// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Routing/Transition should render 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "timeout": 200,
      },
      "attrs": Array [
        [Function],
      ],
      "componentStyle": ComponentStyle {
        "componentId": "Transition__RoutingComponentTransition-sc-2voe5n-0",
        "isStatic": false,
        "rules": Array [
          "position:relative;&.component-enter{opacity:0;transform:scale(0.95);z-index:200;}&.component-enter-active{opacity:1;transform:scale(1);transition:opacity ",
          [Function],
          "ms ease-out,transform ",
          [Function],
          "ms cubic-bezier(0.175,0.885,0.32,1.275);}&.component-exit{opacity:1;transform:scale(1);}&.component-exit-active{opacity:0;height:0 !important;transform:scale(0.95);transition:opacity ",
          [Function],
          "ms ease-out,height ",
          [Function],
          "ms ease-in ",
          [Function],
          "ms,transform ",
          [Function],
          "ms ease-in;}",
        ],
      },
      "displayName": "Transition__RoutingComponentTransition",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "children": [Function],
        "timeout": [Function],
      },
      "render": [Function],
      "styledComponentId": "Transition__RoutingComponentTransition-sc-2voe5n-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  timeout={200}
>
  <div>
    Content
  </div>
</StyledComponent>
`;
